### PR TITLE
Replace Webpack with Vite

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
This PR removed Webpack in favor of Vite. The configuration is much simpler, there are way less dependencies, build times improve significantly and we get hot reload for the index.html file.

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export function getReportingParams(state: PlaybackState): PlaybackProgressInfo {
     return {
         AudioStreamIndex: state.audioStreamIndex,
         CanSeek: state.canSeek,
-        IsMuted: window.volume.muted,
+        IsMuted: window.volume?.muted,
```
Comment (thornbill) [inline src/helpers.ts:46]: This could be undefined, should it be coerced to be a Boolean?

File: src/helpers.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export function getReportingParams(state: PlaybackState): PlaybackProgressInfo {
         PositionTicks: Math.round(getCurrentPositionTicks(state)),
         RepeatMode: window.repeatMode,
         SubtitleStreamIndex: state.subtitleStreamIndex,
-        VolumeLevel: Math.round(window.volume.level * 100)
+        VolumeLevel: Math.round(window.volume?.level * 100)
```
Comment (thornbill) [inline src/helpers.ts:58]: Similarly this could be `undefined * 100`?

**Review by ferferga — COMMENTED** [review]
I believe ts-node, cross-env and source-map-loader are not necessary as well?

**Comment by nielsvanvelzen** [comment]
Yup and I also forgot css-loader

File: vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable sort-keys */
+
+import { defineConfig } from 'vite';
+import { version } from './package.json';
+
+export default defineConfig({
+    root: 'src',
+    build: {
+        outDir: '../dist',
```
Comment (thornbill) [inline vite.config.ts:9]: `../` doesn't seem right here?

Comment (nielsvanvelzen) [reply vite.config.ts:9]: The outDir is relative to the root, our sources are in the "src" folder and we want the dist folder outside of it. Just stupid JavaScript ecosystem things.

Comment (nielsvanvelzen) [reply vite.config.ts:9]: See https://vitejs.dev/config/build-options.html#build-outdir

**Review by thornbill — APPROVED** [review]

**Pull request merged**